### PR TITLE
Fix Rotoscope.trace with non-File IO object and no flatten

### PR DESCRIPTION
**Pull request by dylanahsmith**
## Problem

`Rotoscope.trace(io)` only worked with a File object, since it was just getting the path for the IO object (and some don't even have a path) and writing directly to the file path.  That means that other IO objects like StringIO or Zlib::GzipWriter wouldn't work.

## Solution

I found it hard to follow `Rotoscope.trace`, since it was really an overloaded function, but the code paths were merged using conditionals instead of separated out for clarity.  So I extracted the code paths to separate functions and used private class functions for code re-use.

The actual fix I made was to use a temp file for the intermediate output, similar to when `flatten: true` is used, then the temp file is read in chunks and written to the IO file.

I separated the two code paths in Rotoscope#flatten as well for consistency.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -1,7 +1,7 @@
 PATH
   remote: .
   specs:
-    rotoscope (0.1.0)
+    rotoscope (0.1.1)
```
Comment (dylanahsmith) [inline Gemfile.lock]: I saw this change from just doing `bundle install` without any other changes.  Why do we even check in the Gemfile.lock for a library? Usually it is git ignored.

File: lib/rotoscope.rb [hunk 2]
```diff
@@ -7,43 +7,69 @@
...
+      io_event_trace(dest, blacklist, &block)
     end
 
-    begin
-      rs = Rotoscope.new(dest_file.path, blacklist)
-      rs.trace { yield rs }
+    private
+
+    def with_temp_file(name)
+      temp_file = Tempfile.new("rotoscope_output")
```
Comment (jahfer) [inline lib/rotoscope.rb]: `Tempfile.new(name)`

File: Gemfile.lock (hunk 1 above)
Comment (jahfer) [inline Gemfile.lock]: Good point. I'll add this to the .gitignore separate from this PR.

File: lib/rotoscope.rb [hunk 3]
```diff
@@ -7,43 +7,69 @@
...
+      end
+    end
+
+    def io_event_trace(dest_io, blacklist, &block)
+      temp_event_trace(blacklist, block) do |rs|
+        File.open(rs.log_path) do |rs_file|
+          buffer = String.new
+          until rs_file.eof?
+            rs_file.readpartial(4096, buffer)
+            dest_io.write(buffer)
```
Comment (jahfer) [inline lib/rotoscope.rb]: We can probably use `IO.copy_stream` here instead of writing the logic manually.

Comment (dylanahsmith) [inline lib/rotoscope.rb]: Cool, I didn't know about `IO.copy_stream`
